traits/tests/test_warnings.py

```diff
@@ -0,0 +1,40 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+
+from traits.api import Any, HasTraits
+from traits.has_traits import TraitsWarning
+
+
+class TestTraitWarning(unittest.TestCase):
+
+    def test_invalid_trait_name_prefix(self):
+        with self.assertWarns(TraitsWarning):
+            class UserDefinedClass(HasTraits):
```

Comment:
Since this class and ones below are also defined in `traits`, the warning is not raised.
The following solution is a bit ugly, but I can't think of a better way apart from mangling the code for tests (which I think is worse):
```

from unittest import mock

def mock_module_name(name):
    return mock.patch(__name__ + ".__name__", name)

class TestTraitWarning(unittest.TestCase):

        with self.assertWarns(TraitsWarning):
            with mock_module_name("somepackage"):
                class UserDefinedClass(HasTraits):
                    trait = "something"
```